src/devices_models/devices/thermal_generation.jl

```diff
@@ -187,6 +270,97 @@ function custom_active_power_constraints!(
     end
 end
 
+"""
+This function adds the active power limits of generators. Constraint (17) & (18) from PGLIB
+"""
+function make_active_power_constraints_inputs!(
+    ::Type{<:PSY.ThermalMultiStart},
+    ::Type{<:ThermalMultiStartUnitCommitment},
+    ::Type{<:PM.AbstractPowerModel},
+    feedforward::Nothing,
+    use_parameters::Bool,
+    use_forecasts::Bool,
+)
+    return DeviceRangeConstraintInputs(;
+        range_constraint_inputs = [RangeConstraintInputs(;
+            constraint_name = ACTIVE_RANGE,
+            variable_name = ACTIVE_POWER,
+            limits_func = x -> (
+                min = 0.0,
+                max = PSY.get_activepowerlimits(x).max - PSY.get_activepowerlimits(x).min,
+            ),
+            bin_variable_name = [ON, START, STOP],
+            constraint_func = device_multistart_range,
+            constraint_struct = DeviceMultiStartRangeConstraintsInfo,
+            lag_limits_func = PSY.get_power_trajectory,
+        )],
+    )
+end
+
+function _get_data_for_range_ic(
+    initial_conditions_power::Vector{InitialCondition},
+    initial_conditions_status::Vector{InitialCondition},
+)
+    lenght_devices_power = length(initial_conditions_power)
+    lenght_devices_status = length(initial_conditions_status)
+    @assert lenght_devices_power == lenght_devices_status
+    ini_conds = Matrix{InitialCondition}(undef, lenght_devices_power, 2)
+    idx = 0
+    for (ix, ic) in enumerate(initial_conditions_power)
+        g = ic.device
+        @assert g == initial_conditions_status[ix].device
+        idx += 1
+        ini_conds[idx, 1] = ic
+        ini_conds[idx, 2] = initial_conditions_status[ix]
+    end
+    return ini_conds
+end
+
+"""
+This function adds range constraint for the first time period. Constraint (10) from PGLIB formulation
+"""
+function initial_range_constraints!(
+    psi_container::PSIContainer,
+    devices::IS.FlattenIteratorWrapper{PSY.ThermalMultiStart},
+    model::DeviceModel{PSY.ThermalMultiStart, ThermalMultiStartUnitCommitment},
+    system_formulation::Type{S},
+    feedforward::Union{Nothing, AbstractAffectFeedForward},
+) where {S <: PM.AbstractPowerModel}
+
+    time_steps = model_time_steps(psi_container)
+    resolution = model_resolution(psi_container)
+    key_power = ICKey(DevicePower, PSY.ThermalMultiStart)
+    key_status = ICKey(DeviceStatus, PSY.ThermalMultiStart)
+    initial_conditions_power = get_initial_conditions(psi_container, key_power)
+    initial_conditions_status = get_initial_conditions(psi_container, key_status)
+    ini_conds = _get_data_for_range_ic(initial_conditions_power, initial_conditions_status)
```

Comment:
Do we need to add handling of initial conditions to the common function?

Reply:
Yes, because most constraints other than range have initial conditions